### PR TITLE
feat: Add Gang effect:

### PR DESCRIPTION
**Pull request by SurprisedBird**
- Use the creator as a default target

### CODE REVIEW

File: effects/gang_effect.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            if card_effect: card_effects.append(card_effect)
+
+        for effect in card_effects:
+            effect.deactivate()
+            self.user_interaction.save_active(msg.GangMessages.RESOLVE_SUCCESS)
+
+        return True
+
+    def is_crad_effect(self, effect):
+        is_next_round = (effect.activation_round == self.activation_round + 1)
```
Comment (elias-gvin) [inline effects/gang_effect.py]: misleading var name.
We are cheking if current (not next) round is activation round

Comment (elias-gvin) [reply effects/gang_effect.py]: Also, please change check to
effect.activation_round ==  self.game.round_number

File: effects/gang_effect.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+            if card_effect: card_effects.append(card_effect)
+
+        for effect in card_effects:
+            effect.deactivate()
+            self.user_interaction.save_active(msg.GangMessages.RESOLVE_SUCCESS)
+
+        return True
+
+    def is_crad_effect(self, effect):
```
Comment (elias-gvin) [inline effects/gang_effect.py]: Misleading name.
Please, change to something else.
For example:
`should_be_disabled`
Or
`is_current_round_card_effect`

File: effects/gang_effect.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    def _resolve_impl(self) -> bool:
+        if self.game.round_number == self.activation_round:
+            return False
+
+        card_effects = []
+        citizens = self.context.game.citizens
+
+        for citizen in citizens:
+            card_effect = next(
```
Comment (elias-gvin) [inline effects/gang_effect.py]: Bug.
Discussed

**Review by elias-gvin — CHANGES_REQUESTED** [review]

File: effects/gang_effect.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+        return True
+
+    def is_crad_effect(self, effect):
+        is_next_round = (effect.activation_round == self.activation_round + 1)
+        not_an_action = (not utils.is_action_effect(effect))
+
+        return is_next_round and not_an_action
+
+    def _validate(self, target_number: int) -> bool:
```
Comment (elias-gvin) [inline effects/gang_effect.py]: Should return input status. Not bool
Please, change type hint

File: effects/gang_effect.py (hunk 2 above)
Comment (elias-gvin) [reply effects/gang_effect.py]: Add type hint for return type and for method argument

**Review by elias-gvin — CHANGES_REQUESTED** [review]

File: main.py [hunk 5]
```diff
@@ -1,35 +0,0 @@
-from enum import Enum, auto
-
-from card import Card
```
Comment (elias-gvin) [inline main.py:3]: Why you've removed main.py? 
Please, return the file

**Review by elias-gvin — CHANGES_REQUESTED** [review]

File: effects/gang_effect.py (hunk 1 above)
Comment (SurprisedBird) [reply effects/gang_effect.py]: Done

File: effects/gang_effect.py (hunk 2 above)
Comment (SurprisedBird) [reply effects/gang_effect.py]: Done

File: effects/gang_effect.py (hunk 4 above)
Comment (SurprisedBird) [reply effects/gang_effect.py]: Done

**Comment by elias-gvin** [comment]
Something is wrong with Commit history. Please, check

**Pull request closed without merging**